src/compiler/optimize/optimize-module.ts

```diff
@@ -216,3 +216,64 @@ export const prepareModule = async (
 
   return results;
 };
+
+/**
+ * Merge old and new source maps into one new one.
+ *
+ * If old or new source map value is falsy, return the other as-is.
+ *
+ * Based on the implementation of `merge-source-map`, see:
+ * https://github.com/keik/merge-source-map/blob/master/index.js
+ *
+ * @param oldMap old source map object
+ * @param newMap new source map object
+ * @returns a source map with the new merged into the old
+ */
+async function mergeSourceMaps(oldMap?: RawSourceMap, newMap?: RawSourceMap): Promise<RawSourceMap> {
```

Comment:
this function is basically [copy-pasted from `merge-source-map`](https://github.com/keik/merge-source-map/blob/master/index.js)

Reply:
I just made a few changes to add type annotations and so on